# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

Ranking Pages

This dataset is about content performance and trend direction, so the safest useful lane is ranking pages that are declining and likely need review. I want to focus on a transparent score that suggests which pages a content reviewer should check first.


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

The work improves the decision of what content to review next. The action is a content/SEO reviewer choosing pages for refresh. A wrong recommendation wastes review time and can delay fixing pages that are already declining in search traffic.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [1]:
import pandas as pd

df = pd.read_csv("../../data/raw/content_refresh_anonymized.csv")

print("rows:", len(df))
print("clients:", df["client_id"].nunique())
print("declining pages:", (df["trend_direction"] == "down").sum())

print("\nTrend direction counts:")
print(df["trend_direction"].value_counts())

print("\nImpressions mean / median / std:")
print(df["impressions_90d"].describe()[["mean", "50%", "std"]])

rows: 30000
clients: 32
declining pages: 16262

Trend direction counts:
trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64

Impressions mean / median / std:
mean     5200.366300
50%       731.000000
std     16838.019547
Name: impressions_90d, dtype: float64


The dataset has 30,000 pages from 32 clients. 16,262 pages are labeled trend_direction == 'down', so more than half are declining.
Average impressions over 90 days are about 5200.3663 with a median of 731.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

I can say what the dataset shows: some pages are currently declining in search impressions and some page properties are associated with that decline. I will treat this as a directional, decision-support signal for review prioritization. I will not claim a causal proof, I will not say I am predicting Google’s algorithm, and I will not publish any private page or client identifiers.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.